planet/src/main/webapp/js/itinerary.js

```diff
@@ -26,9 +26,23 @@ window.handleListOptionChange = handleListOptionChange;
 window.addEvent = addEvent;
 window.saveEvents = saveEvents;
 window.generateItinerary = generateItinerary;
+window.initAutocomplete = initAutocomplete; 
 
 const database = firebase.database();
 
+/** Adds autocomplete to input boxes */
+function initAutocomplete() {
+    let startAddress = document.getElementById('starting-address');
+    let eventAddress = document.getElementById('add-event-address');
+
+    let options = {
+        types: ['geocode']
+    };
+
+    let autocompleteStart = new google.maps.places.Autocomplete(startAddress,options); 
```

Comment:
The code here is still confusing, but I guess it's a quirk of the places API.

We're creating an object with `new`, then binding it to a variable. Then we're immediately letting it fall out of scope, removing any reference to the autocomplete object. In theory, the javascript's garbage collector should then delete the object, which would cause autocomplete to stop working. However, I guess that isn't happening, since you've tested this and it works.

More about garbage collection here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Memory_Management

I think the most readable solution would be to define variables on the module scope (by declaring them with `let` outside of any function).
Alternatively, add a comment saying that this is safe because the object doesn't get cleaned up by the garbage collector.

Reply:
Okay! Makes sense